src/pubsub/mod_pubsub.erl

```diff
@@ -796,10 +751,8 @@ disco_items(Host, Node, From) ->
                      Owners = node_owners_call(Host, Type, Nidx, O),
                      case get_allowed_items_call(Host, Nidx, From, Type, Options, Owners) of
                          {result, Items} ->
-                             {result, [#xmlel{name = <<"item">>,
-                                              attrs = [{<<"jid">>, jid:to_binary(Host)},
-                                                       {<<"name">>, ItemId}]}
-                                       || #pubsub_item{itemid = {ItemId, _}} <- Items]};
+                             {result, [disco_item(Host, ItemId) ||
+                                          #pubsub_item{itemid = {ItemId, _}} <- Items]};
```

Comment:
should be nice to have `[pubsub_to_disco_item(Host, Item) || Item Item <- Items]`. So, we would start crashing, if itemid format changes.

Reply:
Good idea, but I think I will limit the changes for now. That code might return items that are not matching the pattern and the filtering might be done on purpose - I don't want to dive deeper into pubsub code now.